### ISSUE TITLE
Connect to Domoticz over MQTT

### ISSUE THREAD

**Issue by djpostka** [omersiar/esp-rfid#443]
Hello,

I would like to connect the esp to Domoticz via MQTT, however, Domoticz works with IDX (identifiers) but the configuration of the ESP does not allow me to enter an IDX.
Do you have a solution? Thank you.

**Comment by matjack1** [comment]
hey @djpostka can you please give me a bit more info on what you would like to do? What should an IDX be? Do you need an extra field on the MQTT message?

**Comment by domoticauser** [comment]
I think he wants to send the RIFD status to Domoticz. Something i also want.

I have a MQTT server and node red. I am not a mqtt expert, but it looks that the RIFD can not send mqtt commands?
Looks like the same question, https://github.com/esprfid/esp-rfid/issues/404

**Comment by djpostka** [comment]
> hey @djpostka can you please give me a bit more info on what you would like to do? What should an IDX be? Do you need an extra field on the MQTT message?

Hello,

Yeah, Domoticz user an extra fiels named "IDX" for the device identification on "Dummy device/virtual device". The domoticez docs explain this: https://www.domoticz.com/wiki/MQTT

**Comment by matjack1** [comment]
thank you @djpostka, I see now. Yes, we would need to add an IDX field for each relay (at least, maybe also for the leds, buttons?) so that you can configure them in esp-rfid and use them in the MQTT messages, right?

I think there needs to be a new page in the settings where you set all the IDXs for all the things that you want to expose and then if the Domoticz integration is enabled, pass the extra IDX in all the MQTT calls.

I'm not familiar with Domoticz and I'm not interested in developing it myself, but I can offer code reviews and suggestions if someone wants to implement it.

**Comment by domoticauser** [comment]
> thank you @djpostka, I see now. Yes, we would need to add an IDX field for each relay (at least, maybe also for the leds, buttons?) so that you can configure them in esp-rfid and use them in the MQTT messages, right?
> 
> I think there needs to be a new page in the settings where you set all the IDXs for all the things that you want to expose and then if the Domoticz integration is enabled, pass the extra IDX in all the MQTT calls.
> 
> I'm not familiar with Domoticz and I'm not interested in developing it myself, but I can offer code reviews and suggestions if someone wants to implement it.

Yes that would be nice! Maybe a idx for each user. The you can see in domoticz who is logging in, and use it for switching lights, warnings, alarm etc..
Idx for Status door sensor, and idx for opening a door.  

Is it possible to let the mqtt send json to node red?

**Comment by matjack1** [comment]
>Is it possible to let the mqtt send json to node red?

Sure! It's already sending MQTT messages on a lot of actions, it's Node Red that should read those messages and do something accordingly.

**Comment by domoticauser** [comment]
> >Is it possible to let the mqtt send json to node red?
> 
> Sure! It's already sending MQTT messages on a lot of actions, it's Node Red that should read those messages and do something accordingly.

I didn't get it at work. MQTT is working with nodered and domoticz,  so that works fine.

**Comment by matjack1** [comment]
Did you configure the MQTT broker correctly? Are you listening to the right MQTT topics?

You should find here: https://github.com/esprfid/esp-rfid/blob/dev/README-ADDON.md the documentation around the MQTT topics that are used, OK? Let me know how it goes!

**Comment by domoticauser** [comment]
Hi,

I think it is correct, maybe i am missing something...
In the MQTT explorer and serial it all looks fine.

Update:, when i delete the topic in the MQTT explorer it gives an error.

![MQTT-Explorer_2022-04-02_12-17-32](https://user-images.githubusercontent.com/11601535/161378782-27fb983f-a4a0-40b9-af32-918099bc336b.jpg)

![firefox_2022-04-02_12-18-48](https://user-images.githubusercontent.com/11601535/161378789-dfeb601c-8644-46f1-bba2-7d87a1e915b9.jpg)


![firefox_2022-04-02_12-22-10](https://user-images.githubusercontent.com/11601535/161378885-1c7954c8-d8fd-4f1b-bee3-f91e9ff827ff.jpg)

![firefox_2022-04-02_17-54-51](https://user-images.githubusercontent.com/11601535/161391020-20c72c43-5f80-448d-91ec-9dc08cf61a9c.jpg)

**Comment by matjack1** [comment]
Hey @marktn I think we are hijacking this issue to something else from the Domoticz integration? :)

Maybe you can open a separate issue where we continue discussing about the Nodered integration. While we are on the topic though, can you help me reproduce your setup so that I can try understanding better how this works? I'm not familiar with Nodered, sorry.

**Comment by domoticauser** [comment]
Yes and no. When MQTT to Node-red works, it can send the information to Domoticz.
I will make a seperate issue! MQTT to node-red.
